src/backend/gpopt/translate/CQueryMutators.cpp

```diff
@@ -702,7 +702,7 @@ CQueryMutators::RunExtractAggregatesMutator(Node *node,
 				//
 				// Note the foo.a var which is in sum() in a subquery must now become a
 				// var referencing the current query level.
-				var->varlevelsup = 0;
+				var->varlevelsup -= context->m_agg_levels_up;
```

Comment:
The comment above is about the previous code. Rewrite please the comment to describe more general case

Reply:
done

Reply:
The comment should be shorter. I suggest to remove the first example. The second example is more general and this one example is enough to understand the algorithm

Reply:
```
If Var references the top level query inside an Aggref that also
references top level query, the Aggref is moved to the derived query
(see comments in Aggref if-case above). Thus, these Var references
are brought up together with the Aggref, i.e.
their varlevelsup are decreased by m_agg_levels_up.
e.g:
select (select max(select foo.a)) from foo;
is transformed into
select (select fnew.max_t)
from (select max(select foo.a) max_t from foo) fnew;
Here the foo.a inside max referenced top level RTE foo at
varlevelsup = 2 inside the Aggref at agglevelsup 1. Then the
Aggref is brought up to the top-query-level of fnew and foo.a
inside Aggref is decreased by original Aggref's level.
```

Reply:
"If Var references the top level query inside an Aggref that also references top level query". The condition on 688th line checks only Var's levels up

Reply:
```
We previously started to mutate the Aggref, that references the top level query.
This Aggref is going to be moved to the derived query (see comments in Aggref if-case above). 
Therefore, if we are mutating Vars inside the Aggref, and these Vars reference the
top level query (varlevelsup = m_current_query_level) as well, we must change their varlevelsup field in
order to preserve correct reference level. i.e these Vars are pulled up as the part of Aggref by the m_agg_levels_up.
e.g:
select (select max(select foo.a)) from foo;
is transformed into
select (select fnew.max_t)
from (select max(select foo.a) max_t from foo) fnew;
Here the foo.a inside max referenced top level RTE foo at
varlevelsup = 2 inside the Aggref at agglevelsup 1. Then the
Aggref is brought up to the top-query-level of fnew and foo.a
inside Aggref is decreased by original Aggref's level.
```